server/src/main/kotlin/delta/codecharacter/server/daily_challenge/DailyChallengeRepository.kt

```diff
@@ -0,0 +1,10 @@
+package delta.codecharacter.server.daily_challenge
+
+import org.springframework.data.mongodb.repository.MongoRepository
+import org.springframework.stereotype.Repository
+import java.util.UUID
+
+@Repository
+interface DailyChallengeRepository : MongoRepository<DailyChallengeEntity, UUID> {
+    fun findByDay(day: Int): DailyChallengeEntity
```

Comment:
This may return nothing when day is negative, so make the return type as ```Optional<DailyChallengeEntity>```. Handle this in ```DailyChallengeService```  properly